core/client/fs/src/main/java/alluxio/client/block/stream/NettyPacketWriter.java

```diff
@@ -220,21 +225,18 @@ public void cancel() {
   @Override
   public void flush() throws IOException {
     mChannel.flush();
-
+    if (mClosed || !mEOFSent || !mCancelSent) {
+      return;
+    }
```

Comment:
I add this condition because of the `FileInstreamIntegration.concurrentRemoteRead` failed without this condition and gave me the `Fail to cache:null` error. It's so strange that `flush()` was called multiple times even when the `close` had been called...

Reply:
The `fs.createFile(), FileOutputStream.write(), FileOutputStream.close() will fail without this line when `WRITE_TYPE=MUST_CACHE` and `USER_SHORT_CIRCUIT_ENABLED=false`
....... Strange......

Reply:
Oh, I make a wrong statement...I will correct it

Reply:
`Flush` will not be called after mClosed, but is probably to be called when EOFSent or CancelSent. Fixed the statement. Thanks